nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/PutBigQueryStream.java

```diff
@@ -0,0 +1,360 @@
+import org.apache.nifi.processor.util.StandardValidators;
+
+/**
+ *
+ */
+
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@Tags({"google", "google cloud", "bq", "bigquery"})
+@CapabilityDescription("Streams flow files to a Google BigQuery table.")
+
+@WritesAttributes({
+    @WritesAttribute(attribute = BigQueryAttributes.DATASET_ATTR, description = BigQueryAttributes.DATASET_DESC),
+    @WritesAttribute(attribute = BigQueryAttributes.TABLE_NAME_ATTR, description = BigQueryAttributes.TABLE_NAME_DESC),
+    @WritesAttribute(attribute = BigQueryAttributes.TABLE_SCHEMA_ATTR, description = BigQueryAttributes.TABLE_SCHEMA_DESC),
+    @WritesAttribute(attribute = BigQueryAttributes.BATCH_SIZE_ATTR, description = BigQueryAttributes.BATCH_SIZE_DESC),
+    @WritesAttribute(attribute = BigQueryAttributes.MAX_ROW_SIZE_ATTR, description = BigQueryAttributes.MAX_ROW_SIZE_DESC),
+    @WritesAttribute(attribute = BigQueryAttributes.CREATE_DISPOSITION_ATTR, description = BigQueryAttributes.CREATE_DISPOSITION_DESC),
+    @WritesAttribute(attribute = BigQueryAttributes.TABLE_CACHE_RESET_ATTR, description = BigQueryAttributes.TABLE_CACHE_RESET_DESC),
+    @WritesAttribute(attribute = BigQueryAttributes.JOB_ERROR_MSG_ATTR, description = BigQueryAttributes.JOB_ERROR_MSG_DESC),
+    @WritesAttribute(attribute = BigQueryAttributes.JOB_ERROR_REASON_ATTR, description = BigQueryAttributes.JOB_ERROR_REASON_DESC),
+    @WritesAttribute(attribute = BigQueryAttributes.JOB_ERROR_LOCATION_ATTR, description = BigQueryAttributes.JOB_ERROR_LOCATION_DESC)
+})
+
+public class PutBigQueryStream extends AbstractBigQueryProcessor {
+    public static final Relationship REL_ROW_TOO_BIG =
+        new Relationship.Builder().name("row_too_big")
+                .description("FlowFiles are routed to this relationship if the row size is too big.")
+                .build();
+
+    public static final PropertyDescriptor DATASET = new PropertyDescriptor
+        .Builder().name(BigQueryAttributes.DATASET_ATTR)
+        .displayName("Dataset")
+        .description(BigQueryAttributes.DATASET_DESC)
+        .required(true)
+        .defaultValue("${" + BigQueryAttributes.DATASET_ATTR + "}")
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .addValidator(StandardValidators.NON_EMPTY_EL_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor TABLE_NAME = new PropertyDescriptor
+        .Builder().name(BigQueryAttributes.TABLE_NAME_ATTR)
+        .displayName("Table Name")
+        .description(BigQueryAttributes.TABLE_NAME_DESC)
+        .required(true)
+        .defaultValue("${" + BigQueryAttributes.TABLE_NAME_ATTR + "}")
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .addValidator(StandardValidators.NON_EMPTY_EL_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor TABLE_SCHEMA = new PropertyDescriptor
```

Comment:
I think `TABLE_SCHEMA`, `DATASET`, `TABLE_NAME` can be moved to the `AbstractBigQueryProcessor`